xc7/tests/ddr/README.md

```diff
@@ -0,0 +1,16 @@
+# UART-DDR test
```

Comment:
Does this test work with Yosys for synthesis into Vivado for place and route?

Reply:
> Does this test work with Yosys for synthesis into Vivado for place and route?

It is worth noting that even if yosys emits a logic invert, Vivado may still remove it during the load of the input.

Reply:
@mithro yes, this work both with Vivado full flow and with Yosys synthesis + Vivado p&r.

For what regards the Yosys + Vivado test, the inverted clock still appears as logic-inverted in the final dcp file (after route).

Reply:
> For what regards the Yosys + Vivado test, the inverted clock still appears as logic-inverted in the final dcp file (after route).

Does this design meet timing?  Comparing the Vivado full flow vs Yosys synthesis, do you see reduced setup and/or hold timing margins?

Reply:
Does the Yosys->Vivado flow work on **hardware**?

Reply:
@mithro Yes it does work, I still need to get a comparison between setup/hold timings though

Reply:
In the after place and route for the Yosys->Vivado flow, what is the pathway for this clock?

Reply:
- Full Vivado flow: the CLKB-related clock for ISERDESes is the same as the CLK one, but, as pointed by @litghost, the CLKB gets automatically inverted.
- Yosys->Vivado flow: the fast clock gets to a SLICE, gets inverted and enters a BUFG, from which it reaches the CLKB pin of the ISERDES.
![Screenshot from 2020-02-04 12-42-37](https://user-images.githubusercontent.com/44773360/73742100-f19dee80-474b-11ea-886a-8e0e80aa888d.png)

The fact is that, by creating some smaller designs with the ISERDES only, and inverting the CLKB with the `IS_CLKB_INVERTED` parameter, nothing actually changes. I have two identical `top.bits` and `top.fasm` of two designs, which I have double checked differ only in the inversion of the CLKB

Reply:
Can you add a picture for the Full Vivado flow too?

Reply:
https://forums.xilinx.com/t5/Implementation/IS-CLK-INVERTED-attribute-of-ISERDES/td-p/991945

> The ISERDESE2 in XAPP524 has “INTERFACE_TYPE = NETWORKING".   Table 3-1 of UG471 says that CLKB should “Always connect to inverted CLK unless in MEMORY_QDR mode.” 
> 
> UG472 shows that outputs of the MMCM come in pairs (eg. CLKOUT1, CLKOUT1B), with CLKOUT1B being an inverted version of CLKOUT1.  So, for the ISERDESE2 in XAPP524, you could (for example) drive the CLK input from MMCM-CLKOUT1 and drive CLKB input from MMCM-CLKOUT1B.  For both CLK and CLKB, ensure that a BUFG is placed between the MMCM and the ISERDESE2 to “balance” the clock routing and reduce skew between the two clocks.  When doing this you should also drive CLKDIV of ISERDESE2 from another clock output of the MMCM (and through a BUFG) as described on page-153 of UG471(v1.10).
>